### DISCUSSION TITLE
Number precision in Quantities

### DISCUSSION THREAD

**Discussion by sebastian-peter** [ie3-institute/simona#292] [General]
Currently, a lot of Quantity numbers are converted to `BigDecimal` during operations like unit conversions or multiplications (cf. profiling of a SIMONA run with mobility sim). Since Quantities are constructed with `double` precision in PSDM, I don't think this is intended for our use case, especially because `BigDecimals` (which are based on `BigInteger`) are significantly slower than doubles during standard operations.

All of this is given narrowing is disabeld, which seems to affect execution speed as well.

**Comment by t-ober** [comment]
I see. Is the conversion taking place in the indriya package? Which would then make it hard to change right?

**Reply by sebastian-peter** [reply]
No, it happens inside `DefaultNumberSystem`, which we partly overwrite anyways

**Comment by sebastian-peter** [comment]
Albeit an being an old article, [this](http://azagorneanu.blogspot.com/2011/07/javamathbigdecimal-performance.html) gives an idea of how `BigDecimal` performes vs. `double`. We should do some testing on our own for Java 17 though, which could be pretty simple ([like this](https://gist.github.com/hillmanli-seekers/a8ab8e463e7b60989e6d0c0900cf4d93)).

**Comment by sebastian-peter** [comment]
This has become obsolete after switching to squants.
